### ISSUE TITLE
CPU issues with ipmi-fan-control 

### ISSUE THREAD

**Issue by chrishoage** [chenxiaolong/ipmi-fan-control#19]
Over time (hours) something  in ipmi-fan-control starts eating cpu causing the system to feel sluggish. While it doesn't eat a lot of actual CPU time (Load average 1.75 1.48. 1.35) opening `htop` and trying to navigate around takes _several seconds_ to navigate between each row. After I kill and restart ipmi-fan-control  my system is fine, and things like htop are immediately more responsive. 

![cpu-usage-over-time](https://user-images.githubusercontent.com/1056886/100903273-8cee1600-347a-11eb-8132-e344e76ee6bf.png)

When I leave htop open for a while, I occasionally start to see zombie processes from smartctl. Could this have anything to do with it? 

![image](https://user-images.githubusercontent.com/1056886/100903600-f2da9d80-347a-11eb-87e0-cab2ab357078.png)

I tried switching from `-A` to `-l scttempsts` to see if the json payload size had anything to do with it, but that does not appear to be the case. 

Anything I can do to debug this issue?

**Comment by chenxiaolong** [comment]
Hmmmm... okay I can definitely reproduce this:
```
[sm-1][~] ps aux | grep 'smartctl.*defunct' | wc -l
529
[sm-1][~] ps aux | grep 'smartctl.*defunct' | wc -l
541
[sm-1][~] ps aux | grep 'smartctl.*defunct' | wc -l
553
...
[sm-1][~] ps aux | grep 'smartctl.*defunct' | wc -l
841
```

I will take a look first thing after work. I suspect I'm using using `std::process::Command` incorrectly. I was assuming that when the `Command` instance goes out of scope, it'll call `waitpid` in the destructor, but that might be a bad assumption.

---

https://doc.rust-lang.org/std/process/struct.Child.html

> The standard library does not automatically wait on child processes (not even if the Child is dropped), it is up to the application developer to do so. As a consequence, dropping Child handles without waiting on them first is not recommended in long-running applications.

Yep, I'm dumb and this should be an easy fix.

**Comment by chenxiaolong** [comment]
Lunch break, so here's a fix: #20 🙂
